File: backend/src/models/Product.js
const { Schema, model } = require('mongoose');

const ProductSchema = new Schema({
    name: { type: String, required: true },
    description: { type: String, required: true },
    price: { type: String, required: true },
    rating: { type: String, required: true, default: '0' },
    available: { type: Boolean, required: true },
    imageURL: { type: String, required: true },
    public_id: { type: String, required: true },
    commerce_id: { type: String, required: true },
    create: { type: Date, default: Date.now }
});

module.exports = model('Product', ProductSchema);

File: app_commerce/lib/Models/Product.dart

class Product {
  //Attributes
  final String id;
  final String name;
  final String description;
  final double price;
  final double rating;
  final bool available;
  final String imageURL;
  final String publicId;
  final String commerceId;

  //Constructor
  Product({
    this.id,
    this.name,
    this.description,
    this.price,
    this.available,
    this.rating,
    this.imageURL,
    this.publicId,
    this.commerceId,
  });

  //Deserialize Json to List<Product>
  factory Product.fromJson(Map<String, dynamic> parsedJson) {
    return Product(
      id: parsedJson['_id'],
      name: parsedJson['name'],
      description: parsedJson['description'],
      price: double.parse(parsedJson['price']),
      available: parsedJson['available'],
      rating: double.parse(parsedJson['rating']),
      imageURL: parsedJson['imageURL'],
      publicId: parsedJson['public_id'],
      commerceId: parsedJson['commerce_id'],
    );
  }
}